src/pfs2tar.c

```diff
@@ -324,14 +587,67 @@ extern void atad_close(void); /* fake_sdk/atad.c */
 int main(int argc, char *argv[])
 {
     int result;
+    bool extract_mode = false;
 
     if (argc < 3) {
         show_help(argv[0]);
         return 1;
     }
 
-    set_atad_device_path(argv[1]);
+    if (strcmp(argv[1], "--extract") == 0)
+        extract_mode = true;
+    else if (strcmp(argv[1], "--restore") == 0)
+        extract_mode = false;
+    else {
+        show_help(argv[0]);
+        return 1;
+    }
+
+    for (int i = 1; i < argc; ++i) {
+        if (strcmp(argv[i], "--overwrite") == 0) {
+            overwrite_mode = true;
+        }
+    }
+
+    // Get the input HDD image path
+    const char *hdd_path = argv[2];
+    const char *last_slash = strrchr(hdd_path, '/');
+    const char *filename = last_slash ? last_slash + 1 : hdd_path;
+
+    // Find the last dot for extension
+    const char *last_dot = strrchr(filename, '.');
+    size_t base_len = last_dot ? (size_t)(last_dot - filename) : strlen(filename);
+    char tar_filename[1024];
+    const char *partition_name = NULL;
+    if (argc >= 4) {
+        if (strncmp(argv[3], "--partition", 11) == 0) {
```

Comment:
Should probably use a strcmp instead of a strncmp to match it exactly instead of just the beginning portion of it.